pkg/controller/splunkforwarder/splunkforwarder_controller.go

```diff
@@ -165,11 +187,13 @@ func (r *ReconcileSplunkForwarder) Reconcile(request reconcile.Request) (reconci
 		}
 	} else if err != nil {
 		return reconcile.Result{}, err
-	} else if instance.CreationTimestamp.After(dsFound.CreationTimestamp.Time) || updateDaemonSet {
-		err = r.client.Update(context.TODO(), daemonSet)
+	} else if instance.CreationTimestamp.After(dsFound.CreationTimestamp.Time) || updateDaemonSet || r.CheckGenerationVersionOlder(dsFound.GetAnnotations(), instance) {
+		err = r.client.Delete(context.TODO(), daemonSet)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+		// Requeue to create the daemonset
+		return reconcile.Result{RequeueAfter: 1}, nil
```

Comment:
Is there a reason you are requeuing after 1 second? You could just `{Requeue: true}`

Reply:
oh, yeah i can do that.. fixed